src/main/java/com/codeus/winter/config/DefaultBeanFactory.java

```diff
@@ -457,7 +462,15 @@ protected Object resolveBean(Class<?> beanClass) {
         if (candidates.size() == 1) {
             targetCandidate = candidates.getFirst();
         } else {
-            //TODO #35, #46: there are multiple candidates, add logic to choose one based on @Primary, @Qualifier or other util annotation.
+            String suggestedName = qualifierAnnotationAutowireCandidateResolver.getSuggestedName(descriptor);
+            if (suggestedName != null) {
+                for (Map.Entry<String, BeanDefinition> candidate : candidates) {
+                    if (candidate.getKey().equals(suggestedName)) {
+                        return getBean(candidate.getKey(), candidate.getValue());
+                    }
+                }
+            }
```

Comment:
I believe the same logic should be supported by our `AutowiredAnnotationBeanPostProcessor`.
Adjustment seems simple:
1. make `com.codeus.winter.config.AbstractAutowireCapableBeanFactory#resolveDependency` public;
2. use `AbstractAutowireCapableBeanFactory` instead of `BeanFactory` in the `AutowiredAnnotationBeanPostProcessor`;
3. replace `beanFactory.getBean(...)` calls with `beanFactory.resolveDependency(...)` in `AutowiredAnnotationBeanPostProcessor`.

Also, you can add a convenient static method for the `DependencyDescriptor` to create it from `java.lang.reflect.Parameter` and use that method in both `ContructorResolver` and `AutowiredAnnotationBeanPostProcessor`.

Please, rebase before making these changes.

Reply:
Thanks, fixed